### ISSUE TITLE
[false-negative] The Checker Framework annotations have no effect when used on types (ElementType.TYPE_USE)

### ISSUE THREAD

**Issue by mewore** [spotbugs/spotbugs#1657]
Ideally, this code should produce a warning:
```java
public class Issue1657 {
    public @org.checkerframework.checker.nullness.qual.NonNull String testReturnNullFromListInNonNullMethod() {
        final List<@org.checkerframework.checker.nullness.qual.Nullable String> nullableStringList = new ArrayList<>();
        nullableStringList.add(null);
        return nullableStringList.get(0); // expected NP_NONNULL_RETURN_VIOLATION
    }
}
```
but it does not.

This might be a new feature, not a bug, since it's probably tricky to implement, or maybe even impossible to achieve by only analyzing the bytecode.

Regardless, I would expect that if a type is annotated as nullable, then SpotBugs should consider all of its return values and variables to be possibly null. Otherwise there should be a disclaimer in the description of warnings like [`NP_NONNULL_RETURN_VIOLATION`](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#np-method-may-return-null-but-is-declared-nonnull-np-nonnull-return-violation) so that SpotBugs users know what to expect.

Here is the test ([commit](https://github.com/mewore/spotbugs/commit/89fff05256f5de6a8aaa1cf831fb72c6798a09e7)):
- https://github.com/mewore/spotbugs/blob/issue-1657/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/nullness/Issue1657Test.java
- https://github.com/mewore/spotbugs/blob/issue-1657/spotbugsTestCases/src/java/ghIssues/Issue1657.java
    - :warning: I have added an extra method testReturnNullInNonNullMethod as a control case to show that the Checker Framework NonNull annotation itself works.

And the error (only the lower method has caused a warning so there's only 1 instead of 2):
```
Expected: Iterable containing exactly <2> BugInstance with:
bugType="NP_NONNULL_RETURN_VIOLATION",
     but: was <[NP: Method may return null, but is declared @Nonnull]>
```

**Comment by KengoTODA** [comment]
FindBugs was implemented when Java has no generics nor `TYPE_USE` so basically SpotBugs does not support it.
We're waiting for contributions from the community.

I personally want to solve this problem when [standard Java annotations for static analysis](http://jspecify.dev/) becomes stable, by implementing a [SpotBugs plugin based on it](https://github.com/spotbugs/spotbugs-jspecify-plugin). Still not sure when it'll be stable, though.
